### ISSUE TITLE
GSoD: Google developer documentation style guide and American English for our docs

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10608]
Discussed as part of our [Create Wagtail developer onboarding tutorials project](https://github.com/wagtail/wagtail/discussions/10606) and subsequently discussed in a core team meeting. We want to switch from British English to American English for our documentation, and introduce the [Google developer documentation style guide](https://developers.google.com/style/). We hope having a defined style guide will make it simpler to write high-quality documentation, and using American English will work better for our contributors. 

 Copy of @activus-d analysis in his documentation audit:

> […] I propose a switch to the American spelling style. This is because the documentation's target audience is global, and most English speakers are more accustomed to the American spelling style than the British.
> To avoid and correct many issues in the Wagtail developer documentation, it's important to implement a writing style guide consistently across all sections. This will ensure consistency in grammar, tone, and language.
> During the Outreachy program, we conducted a review to identify the most suitable writing style guide for the Editor guide documentation project. The [unanimous decision was to adopt the Google developer documentation style guide](https://github.com/wagtail/guide/discussions/282#discussioncomment-4332158). It would be beneficial to apply the same guide to the Wagtail developer documentation for the same reasons. Additionally, we have the flexibility to customize the writing style by including specific exceptions or additions to the rules outlined in the Google developer documentation style guide within the Documentation guidelines section.

**Comment by thibaudcolas** [comment]
@activus-d when you have the chance, could you refine this to have more specific recommendations / tasks on how we transition to the new style guide for our whole documentation?

See [this comment from @lb-](https://github.com/wagtail/wagtail/pull/10579#discussion_r1238022760):

> Ok. I do recall seeing something about this, our current docs say we should not spell things this way so we should probably update our own documentation to reflect this decision. 
> 
> https://docs.wagtail.org/en/stable/contributing/general_guidelines.html#language still says British English. 
> 
> I'd also recommend we then update the docs spell checker, and maybe a bulk update. 
> 
> Maybe this spelling change needs its own issue created so there can be a proper plan in place. 
> 
> Even as recently as last month I updated some spelling and PRs to British spelling.

**Comment by lb-** [comment]
Thanks for the ping, I had not seen this issue pop up.

**Comment by thibaudcolas** [comment]
cc @vossisboss who also said as part of Damilola’s audit:

> Additionally, I feel like The Chicago Manual of Style would be the most appropriate style guide for us to adopt for non-technical communications.

**Comment by thibaudcolas** [comment]
@activus-d will work on this after #10579, starting with documenting our style guide and running Vale over the documentation. We’ll document this most likely in either of the following pages:

- https://docs.wagtail.org/en/stable/contributing/documentation_modes.html
- https://docs.wagtail.org/en/stable/contributing/documentation_guidelines.html

We also need to document which exceptions we allow ourselves to make (exclamation marks).

**Comment by lb-** [comment]
We will need to.make it clear when to use UK VA US English. 

1. Code documentation (docstrings, comments)
2. Main documentation
3. UI default language (i.e. default translation)

Currently our documentation aludes to all three being the same, we may want to be clear about what's expected. I assume we want to keep the default UI language as UK, but not sure, we may want to update some translation strings also.

Obviously, some conent matters more than others, along with some spelling specifics.

https://docs.wagtail.org/en/stable/contributing/general_guidelines.html#language

**Comment by lb-** [comment]
Might be good to revisit this old PR (just closed) that added a spell checker to the CI.
https://github.com/wagtail/wagtail/pull/7412

**Comment by activus-d** [comment]
as @thibaudcolas suggested in [PR #10634](https://github.com/wagtail/wagtail/pull/10634#discussion_r1255466725), here are the instructions for using Vale to lint documentation and ensure consistency with the rules stated in the Google Developer Style Guide:

> Please note that although Vale is helpful for identifying inconsistencies in the style guide, some of the writing style rules in the Google Developer Documentation Style Guide have exemptions. Therefore, contributors must exercise their discretion to determine whether the inconsistencies flagged by Vale qualify as exemptions. To make the most of Vale, contributors need to ensure they have read the Google Developer Style Guide thoroughly and understand both the general rules and their exemptions.


## Writing style linter

Before submitting a pull request, use [Vale](https://vale.sh/) to check if your documentation adheres to the Google developer documentation style guide. Follow these steps to check your documentation with Vale:

1.  Copy and paste your Wagtail documentation content into a new markdown file located in a separate directory.
2.  Open your command line terminal and navigate to the separate directory that has your new markdown file.
3.  Install Vale using a package manager suitable for your operating system, such as [Homebrew](https://brew.sh/) for macOS or [Chocolatey](https://phoenixnap.com/kb/chocolatey-windows) for Windows.
    -  For Windows:
    
    ```doscon
    choco install vale
    ```
    
    -  For macOS:
    
    ```sh
    brew install vale
    ```

4.  Verify the successful installation of Vale by running the command:

```sh
vale -v
```

This should display the version of Vale you installed.
5.  Create a configuration file named `.vale.ini` in the same directory.
6.  Paste the following contents into your `.vale.ini` configuration file:

```txt
StylesPath = styles
MinAlertLevel = suggestion
Packages = Google

[*]
BasedOnStyles = Vale, Google
```

7.  Initialize your directory by running the command:

```sh
vale sync
```

8.  Check for writing style errors in your documentation:

```sh
vale <documentation.md>
```

```{note}
Replace "documentation.md" with the actual name of your markdown file.
```

If you follow the preceding steps correctly, your command line terminal will display errors indicating non-adherence to the Google developer documentation style guide. To learn more about setting up Vale, refer to the [Vale documentation](https://vale.sh/docs/).
